examples/ai-core/src/generate-text/openai-log-metadata-middleware.ts

```diff
@@ -0,0 +1,35 @@
+import { openai } from '@ai-sdk/openai';
+import {
+  generateText,
+  Experimental_LanguageModelV1Middleware as LanguageModelV1Middleware,
+  experimental_wrapLanguageModel as wrapLanguageModel,
+} from 'ai';
+import 'dotenv/config';
+
+const logProviderMetadataMiddleware: LanguageModelV1Middleware = {
+  transformParams: async ({ params }) => {
+    console.log(
+      'providerMetadata: ' + JSON.stringify(params.providerMetadata, null, 2),
+    );
+    return params;
+  },
+};
+
+async function main() {
+  const { text } = await generateText({
+    model: wrapLanguageModel({
+      model: openai('gpt-4o'),
+      middleware: logProviderMetadataMiddleware,
+    }),
+    experimental_providerMetadata: {
+      myMiddleware: {
```

Comment:
where does `myMiddleware` name come from?

Reply:
you can enter anything here